pkg/controller/core/workload_controller.go

```diff
@@ -826,9 +826,20 @@ func (r *WorkloadReconciler) admittedNotReadyWorkload(wl *kueue.Workload) (bool,
 	if podsReadyCond != nil && podsReadyCond.Status == metav1.ConditionTrue {
 		return false, 0
 	}
-	admittedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadAdmitted)
-	elapsedTime := r.clock.Since(admittedCond.LastTransitionTime.Time)
-	return true, max(r.waitForPodsReady.timeout-elapsedTime, 0)
+
+	if podsReadyCond == nil || podsReadyCond.Reason == kueue.WorkloadWaitForPodsStart {
+		admittedCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadAdmitted)
+		elapsedTime := r.clock.Since(admittedCond.LastTransitionTime.Time)
+		return true, max(r.waitForPodsReady.timeout-elapsedTime, 0)
+	} else {
+		// podsReadyCond.Reason == kueue.WorkloadWaitForPodsRecovery
+		// A pod has failed and the workload is waiting for recovery
```

Comment:
What if `podReadyCond` has `reason: PodsReady` and `status: False`?

Reply:
With the current flow `podsReadyCond` with `PodsReady` reason will be converted into `WaitForPodsStart` here:
https://github.com/kubernetes-sigs/kueue/pull/4301/files#diff-552ab657667de70900b502e2617bcb405fbf4203ff08b2a35d7c87f2d637a75bR1196-R1202